interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -359,28 +359,31 @@ public void pingPong() throws Exception {
             .build());
 
     @SuppressWarnings("unchecked")
-    StreamObserver<StreamingOutputCallResponse> responseObserver = mock(StreamObserver.class);
+    StreamRecorder<StreamingOutputCallResponse> responseObserver = StreamRecorder.create();
     StreamObserver<StreamingOutputCallRequest> requestObserver
         = asyncStub.fullDuplexCall(responseObserver);
     for (int i = 0; i < requests.size(); i++) {
       requestObserver.onNext(requests.get(i));
-      verify(responseObserver, timeout(operationTimeoutMillis())).onNext(goldenResponses.get(i));
-      verifyNoMoreInteractions(responseObserver);
     }
+
     requestObserver.onCompleted();
-    verify(responseObserver, timeout(operationTimeoutMillis())).onCompleted();
-    verifyNoMoreInteractions(responseObserver);
+    responseObserver.awaitCompletion();
+    assertSuccess(responseObserver);
+    assertEquals(goldenResponses.size(), responseObserver.getValues().size());
+    for (int ix = 0; ix < responseObserver.getValues().size(); ++ix) {
+      StreamingOutputCallResponse response = responseObserver.getValues().get(ix);
+      assertEquals(goldenResponses.get(ix), response);
+    }
   }
 
   @Test(timeout = 10000)
   public void emptyStream() throws Exception {
     @SuppressWarnings("unchecked")
```

Comment:
Remove the `@SuppressWarnings`. Ditto elsewhere (where you removed the `mock()`).